runtime/hsa-runtime/core/runtime/amd_topology.cpp

```diff
@@ -68,58 +72,42 @@ extern r_debug _amdgpu_r_debug;
 
 namespace rocr {
 namespace AMD {
-// Minimum acceptable KFD version numbers
-static const uint kKfdVersionMajor = 0;
-static const uint kKfdVersionMinor = 99;
-
-void DiscoverDrivers(bool &gpu_found, bool &aie_found) {
-  // Open connection to GPU and AIE kernel drivers.
-  gpu_found = (KfdDriver::DiscoverDriver() == HSA_STATUS_SUCCESS);
-  aie_found = (XdnaDriver::DiscoverDriver() == HSA_STATUS_SUCCESS);
-}
+// Anonymous namespace.
+namespace {
+const std::vector<std::function<hsa_status_t(std::unique_ptr<core::Driver> &)>>
+    discover_driver_funcs = {
+#ifdef __linux__
+        KfdDriver::DiscoverDriver, XdnaDriver::DiscoverDriver
+#endif
+};
 
-// Query for user preference and use that to determine Xnack mode of ROCm system.
-// Return true if Xnack mode is ON or false if OFF. Xnack mode of a system is
-// orthogonal to devices that do not support Xnack mode. It is legal for a
-// system with Xnack ON to have devices that do not support Xnack functionality.
-bool BindXnackMode() {
-  // Get users' preference for Xnack mode of ROCm platform
-  HSAint32 mode;
-  mode = core::Runtime::runtime_singleton_->flag().xnack();
-  bool config_xnack =
-      (core::Runtime::runtime_singleton_->flag().xnack() != Flag::XNACK_REQUEST::XNACK_UNCHANGED);
-
-  // Indicate to driver users' preference for Xnack mode
-  // Call to driver can fail and is a supported feature
-  HSAKMT_STATUS status = HSAKMT_STATUS_ERROR;
-  if (config_xnack) {
-    status = hsaKmtSetXNACKMode(mode);
-    if (status == HSAKMT_STATUS_SUCCESS) {
-      return mode;
-    }
-  }
+void DiscoverDrivers() {
+  for (auto discover_driver_fn : discover_driver_funcs) {
```

Comment:
```suggestion
  for (const auto &discover_driver_fn : discover_driver_funcs) {
```
to avoid copying the std::function objects.

Reply:
Fixed